src/code/index.js

```diff
@@ -118,10 +129,49 @@ function previewLink(button, value) {
     button.addEventListener("click", () => {
         window.open(inputValue, "_blank");
     });
-
+    
     // Rest of the code...
+    let matchingKey;
+    const matchingLink = Object.entries(socialLinks).find(([key, value]) => {
+        const url = new URL(value.replace("<username>", ""));
+        const domainKeyWords = url.hostname.replace("www.", "").replace(".com", "");
```

Comment:
Hey,
See all URLs doesn't have `www`, please go through the `social-links.json`, and see the format of URLs, it should be generalized for all types of URLs.
For few it is working, but not for all.

Reply:
Hey, 
I added in handling for the socialLinks values that start with the same domain keywords such as:
"dev.to", "devpost.com", "devfolio.co" etc.

I was stuck on how retrieve the key as soon as the minimum keyword is entered like  `https://www.linkedin` while also not conflicting with the "dev" URLs but I think this code should handle it and any URLs in the future that start with the same domain keywords.